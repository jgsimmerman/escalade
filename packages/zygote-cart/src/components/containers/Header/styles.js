import css from 'styled-jsx/css';

export default css`
  .zygoteClose {
    color: #000;
    font-size: 3em;
    font-weight: 200;
    position: absolute;
    top: -10px;
    right: 20px;
    cursor: pointer;
  }
  .zygoteHeader {
    text-align: center;
    padding: 20px 20px 0px 20px;
    position: relative;
  }
  .zygoteBrandLogo > img {
    margin: 0;
    padding: 0;
    width: 200px;
  }
  .zygoteBrandLogo {
    padding-bottom: 15px;
  }
  .zygoteDefaultBrandLogo {
    background: #ccc;
    padding: 5px 0;
    text-transform: uppercase;
    width: 120px;
    margin: 0 auto;
    font-weight: 300;
    color: #191919;
  }
  .zygoteAddedToCart {
    font-size: 1.2em;
    font-weight: 300;
    font-style: italic;
    margin-bottom: 0.8em;
  }
  .zygoteCartHeader {
    margin-bottom: 20px;
  }
`;
