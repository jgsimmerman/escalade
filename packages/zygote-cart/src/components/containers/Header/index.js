import React, { Component } from 'react'
import Cookie from 'js-cookie'
import { Subscribe } from 'statable'

import { cartState, itemState, cost, userInfo } from '../../state'
import { Nav } from '../../containers'
import { closeCart, resetCart } from '../../../injectState'
import { upperFirst } from '../../utils'
import styles from './styles.js'

export default class Header extends Component {
  constructor(props) {
    super(props)

    this.checkState = this.checkState.bind(this)
  }

  checkState() {
    closeCart()
    const { tab, loading, apiErrors, errors } = cartState.state
    cartState.setState({ tab: 0 })

    if (tab === 3 && loading === false) {
      if (errors || apiErrors) {
        cartState.setState({ tab: 0 })
        return
      }
      resetCart()
    }
  }

  render() {
    return (
      <Subscribe to={[cartState, itemState]}>
        {(state, itemObj) => (
          <div className="zygoteHeader">
            <div className="zygoteClose" onClick={() => this.checkState()}>
              Ã—
            </div>
            <div className="zygoteBrandLogo">
              {this.props.brandLogo ? (
                <img src={this.props.brandLogo} alt="" />
              ) : (
                <div className="zygoteDefaultBrandLogo">Brand Logo</div>
              )}
            </div>
            {itemObj.items.length === 0 ? null : state.tab === 1 ? (
              <Nav />
            ) : state.tab === 2 ? (
              <Nav />
            ) : state.tab === 3 ? (
              <div>
                <div className="zygoteOrderComplete">
                  <div className="zygoteOrderCompleteTitle">
                    {this.props.orderCompleteTitle}
                  </div>
                  <div className="zygoteOrderCompleteBody">
                    {this.props.orderCompleteBody ||
                      `Your ${upperFirst(state.site)} order is in!`}
                  </div>
                </div>
              </div>
            ) : (
              <div>
                <div className="zygoteAddedToCart">
                  {this.props.addToCartMessage}
                </div>
                {this.props.cartHeader ? (
                  <div className="zygoteCartHeader">
                    {this.props.cartHeader}
                  </div>
                ) : null}
              </div>
            )}

            <style jsx>{styles}</style>
          </div>
        )}
      </Subscribe>
    )
  }
}

Header.defaultProps = {
  addToCartMessage: "You've added to your cart!",
  orderCompleteTitle: 'Yessss!'
}
