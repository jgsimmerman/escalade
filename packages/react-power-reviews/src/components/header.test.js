import React from "react"
import renderer from "react-test-renderer"

import Header from "./header"

describe(`Header`, () =>
	it(`renders correctly`, () => {
		const container = renderer.create(<Header />)
		const tree = container.toJSON()
		expect(tree).toMatchSnapshot()
	})
)