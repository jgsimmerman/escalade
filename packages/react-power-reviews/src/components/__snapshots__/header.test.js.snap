// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="css-17qhf9j"
>
  <ul>
    <li>
      <Link
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        to="/about"
      >
        About
      </Link>
    </li>
    <li>
      <Link
        to="/grid"
      >
        Grid
      </Link>
    </li>
    <li>
      <Link
        to="/contact"
      >
        Contact
      </Link>
    </li>
    <li>
      <a
        href="/email-templates/invitation"
      >
        Email Template
      </a>
    </li>
  </ul>
</header>
`;
